plugins/modules/rabbitmq_user_limits.py

```diff
@@ -0,0 +1,219 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2013, Chatham Financial <oss@chathamfinancial.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: rabbitmq_user_limits
+short_description: Manage RabbitMQ user limits
+description:
+  - Manage the state of user limits in RabbitMQ
+author: Aitor Pazos (@aitorpazos)
+options:
+  user:
+    description:
+      - Name of user to manage limits for
+    type: str
+    required: true
+    aliases: [username, name]
+  max_connections:
+      description:
+          - Max number of concurrent client connections.
+          - Negative value means "no limit".
+          - Ignored when the I(state) is C(absent).
+      type: int
+      default: -1
+  max_channels:
+      description:
+          - Max number of channels.
+          - Negative value means "no limit".
+          - Ignored when the I(state) is C(absent).
+      type: int
+      default: -1
+  node:
+      description:
+          - Name of the RabbitMQ Erlang node to manage.
+      type: str
+  state:
+      description:
+          - Specify whether the limits are to be set or cleared.
+          - If set to C(absent), the limits of both I(max_connections) and I(max_channels) will be cleared.
+      type: str
+      default: present
+      choices: [present, absent]
+'''
+
+EXAMPLES = '''
+# Limit both of the max number of connections and channels on the user 'guest'.
+- community.rabbitmq.rabbitmq_user_limits:
+    user: guest
+    max_connections: 64
+    max_channels: 256
+    state: present
+
+# Limit the max number of connections on the user 'guest'.
+# This task implicitly clears the max number of channels limit using default value: -1.
+- community.rabbitmq.rabbitmq_user_limits:
+    user: guest
+    max_connections: 64
+    state: present
+
+# Clear the limits on the user 'guest'.
+- community.rabbitmq.rabbitmq_user_limits:
+    user: guest
+    state: absent
+'''
+
+RETURN = ''' # '''
+
+import json
+import re
+from distutils.version import LooseVersion as Version
+from ansible.module_utils.basic import AnsibleModule
+
+
+class RabbitMqUserLimits(object):
+    def __init__(self, module):
+        self._module = module
+        self._max_connections = module.params['max_connections']
+        self._max_channels = module.params['max_channels']
+        self._node = module.params['node']
+        self._state = module.params['state']
+        self._user = module.params['user']
+        self._rabbitmqctl = module.get_bin_path('rabbitmqctl', True)
+
+        self._version = self._rabbit_version()
+
+    def _exec(self,
+              args,
+              run_in_check_mode=False,
+              split_lines=True):
+
+        if not self._module.check_mode or (self._module.check_mode and run_in_check_mode):
```

Comment:
```suggestion
              run_in_check_mode=False,
              split_lines=True):

        if not self._module.check_mode or (self._module.check_mode and run_in_check_mode):
```
is the `run_in_check_mode` arg really necessary? For me personally it looks a bit confusing?

Reply:
It is a bit confusing to me as well, but I wanted to keep the [pattern used in other modules](https://github.com/ansible-collections/community.rabbitmq/blob/48947b67c000f356f757ad8879aa9f760e46b284/plugins/modules/rabbitmq_vhost.py#L65) so this one wouldn't feel too alien.
Functionally it is needed in order to always run the ` _rabbit_version` regardless of the check mode value.

Reply:
okay but would be nice to refactor it somehow if possible. Can be hard for further folks to figure out what's going on and why.

Reply:
I will open a separate PR renaming `run_in_check_mode` to `force_exec_in_check_mode` which should make it much easier to understand.